spyder_notebook/notebookplugin.py

```diff
@@ -166,13 +168,26 @@ def get_plugin_actions(self):
 
     def register_plugin(self):
         """Register plugin in Spyder's main window."""
-        self.focus_changed.connect(self.main.plugin_focus_changed)
-        self.main.add_dockwidget(self)
-        self.create_new_client(give_focus=False)
-        icon_path = os.path.join(PACKAGE_PATH, 'images', 'icon.svg')
-        self.main.add_to_fileswitcher(self, self.tabwidget, self.clients,
-                                      QIcon(icon_path))
-        self.recent_notebook_menu.aboutToShow.connect(self.setup_menu_actions)
+        if self.check_compatibility()[0]:

```

Comment:
This is an error in Spyder then. I'll open an issue so you can help me to fix it.